packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1210,7 +1210,9 @@ class WidgetsApp extends StatefulWidget {
     _debugShowWidgetInspectorOverrideNotifier.value = value;
   }
 
-  static final ValueNotifier<bool> _debugShowWidgetInspectorOverrideNotifier = ValueNotifier<bool>(false);
+  static final ValueNotifier<bool> _debugShowWidgetInspectorOverrideNotifier =
+    FlutterMemoryAllocations.instance.exemptFromDisposal(() => ValueNotifier<bool>(false));
```

Comment:
How is it ensured that in release builds with memory tracking disabled all this unused machinery is tree-shaken out so performance and binary size is not impacted?

Reply:
This also adds a lot of clutter to the code making something really simple and straight forward harder to read.

Reply:
Moved out from FlutterMemoryAllocations and added condition.
It is little nicer now.